application/src/main/java/run/halo/app/theme/endpoint/CommentFinderEndpoint.java

```diff
@@ -152,7 +165,37 @@ Mono<ServerResponse> createComment(ServerRequest request) {
                 comment.getSpec().setUserAgent(HaloUtils.userAgentFrom(request));
                 return commentService.create(comment);
             })
-            .flatMap(comment -> ServerResponse.ok().bodyValue(comment));
+            .transformDeferred(createIpBasedRateLimiter(request))
+            .flatMap(comment -> ServerResponse.ok().bodyValue(comment))
+            .onErrorResume(RequestNotPermitted.class, e -> ServerResponse.status(TOO_MANY_REQUESTS)
+                .contentType(MediaType.APPLICATION_JSON)
+                .bodyValue(createErrorResponse(e,
+                    TOO_MANY_REQUESTS,
+                    REQUEST_NOT_PERMITTED_TYPE,
+                    request.exchange())));
+    }
+
+    private ErrorResponse createErrorResponse(Throwable t, HttpStatus status, String type,
+        ServerWebExchange exchange) {
+        var errorResponse = ErrorResponse.create(t, status, t.getMessage());
+        var problemDetail = errorResponse.updateAndGetBody(messageSource, getLocale(exchange));
+        problemDetail.setType(URI.create(type));
+        problemDetail.setInstance(exchange.getRequest().getURI());
+        problemDetail.setProperty("requestId", exchange.getRequest().getId());
+        problemDetail.setProperty("timestamp", Instant.now());
+        return errorResponse;
+    }
+
+    private Locale getLocale(ServerWebExchange exchange) {
```

Comment:
The method was unused if we removed the error handler above.

Reply:
but i get the following error if i remove the error handler
```
2023-06-25T15:27:08.288+08:00 ERROR 23128 --- [ctor-http-nio-2] a.w.r.e.AbstractErrorWebExceptionHandler : [42a68b54-830]  500 Server Error for HTTP POST "/apis/api.halo.run/v1alpha1/comments"

io.github.resilience4j.ratelimiter.RequestNotPermitted: RateLimiter 'comment-creation-from-ip-0:0:0:0:0:0:0:1' does not permit further calls
	at io.github.resilience4j.ratelimiter.RequestNotPermitted.createRequestNotPermitted(RequestNotPermitted.java:43) ~[resilience4j-ratelimiter-2.0.2.jar:2.0.2]
```

Reply:
If we don't want to see the stack trace, property `resilience4j.ratelimiter.configs.comment-creation.writableStackTraceEnabled` (default is true) should be set to `false`.

Reply:
An alert like this will also be displayed in the browser. Is that okay?
![image](https://github.com/halo-dev/halo/assets/73240868/462e9de4-3cf0-4b3b-8608-a72339dc4364)

Reply:
IMO, the type of problem detail is correct, but the status should be `429` instead of `500`. I will submit another PR to fix the problem. Thanks for reminding me!

```json
{
  "type": "https://halo.run/probs/request-not-permitted",
  "title": "Internal Server Error",
  "status": 500,
  "detail": "Something went wrong, please try again later.",
  "instance": "http://localhost:8090/apis/api.halo.run/v1alpha1/comments",
  "requestId": "9877ddb2-4",
  "timestamp": "2023-06-25T15:16:19.267508Z"
}
```